### PR TITLE
Refactor loadbal order-options

### PR DESCRIPTION
**Pull request by caberos**
https://github.com/softlayer/softlayer-python/issues/1520

### CODE REVIEW

**Review by ATGE — COMMENTED** [review]
the command `slcli lb order-options` works well, just the message displayed as an error looks weird.
Also, the output of the `slcli lb order-options --datacenter <DC>` in the worked branch is the same output as the latest `slcli` version shows, so looks like the table requirement mentioned in the issue, is not implemented. #1520 

> `slcli lb order-options -d dal10` works fine, but should output 2 tables, instead of 1 table with a table in each column.
> 
> Table 1 titled " Prices", Table 2 titled " Subnets Private Subnets"
### Testing
 - ✔️ slcli loadbal order-options
 
 
image

- slcli loadbal order-options -d dal13
 worked branch output:
image
 slcli (SoftLayer Command-line) v5.9.6 output:
image

File: SoftLayer/CLI/loadbal/order.py [hunk 1]
```diff
@@ -89,53 +89,64 @@ def order_options(env, datacenter):
...
+    if not datacenter:
+        data_table = formatting.KeyValueTable(['Datacenters', 'City'])
+        for region in package['regions']:
+            data_table.add_row([region['description'].split('-')[0], region['description'].split('-')[1]])
+            # print(region)
+        env.fout(data_table)
+        click.secho("ERROR: Use `slcli lb order-options --datacenter <DC>` "
+                    "to find pricing information and private subnets for that specific site.")
+
+    else:
```
[the remark this answers is not in the document]
Comment (allmightyspiff) [reply SoftLayer/CLI/loadbal/order.py:101]: Most of this logic is just table formatting, so I'm ok with it not being in a manager specifically, unless we want to use it in more than just this one command. Breaking it up into a few smaller sub-functions that return `Table` objects would make this logic a bit more readable though.

File: SoftLayer/CLI/loadbal/order.py [hunk 2]
```diff
@@ -89,53 +89,64 @@ def order_options(env, datacenter):
...
+                # Only show these types, easier to filter here than in an API call.
+                if subnet.get('subnetType') != 'PRIMARY' and \
+                        subnet.get('subnetType') != 'ADDITIONAL_PRIMARY':
+                    continue
+                space = "{}/{}".format(subnet.get('networkIdentifier'), subnet.get('cidr'))
+                vlan = "{}.{}".format(subnet['podName'], subnet['networkVlan']['vlanNumber'])
+                subnet_table.add_row([subnet.get('id'), space, vlan])
+            this_table.add_row([price_table, subnet_table])
+
+            env.fout(this_table)
```
Comment (allmightyspiff) [inline SoftLayer/CLI/loadbal/order.py:149]: I'd like this to be 2 tables now, so the following should produce the output I'm looking for.

```suggestion
            env.fout(price_table)
            env.fout(subnet_table)
```

**Review by allmightyspiff — CHANGES_REQUESTED** [review]
Along with my comments, I'd like to see a unit test specifically for this command with a location specified.

File: SoftLayer/CLI/loadbal/order.py [hunk 3]
```diff
@@ -89,53 +89,60 @@ def order_options(env, datacenter):
...
+        env.fout(data_table)
+        click.secho("Use `slcli lb order-options --datacenter <DC>` "
+                    "to find pricing information and private subnets for that specific site.")
+
+    else:
+        for region in package['regions']:
+            dc_name = utils.lookup(region, 'location', 'location', 'name')
+
+            # Skip locations if they are not the one requested.
+            if datacenter and dc_name != datacenter:
```
Comment (allmightyspiff) [inline SoftLayer/CLI/loadbal/order.py:106]: The datacenter lists the DC name as all uppercase (`DAL13`), but if you search for `DAL13`, this command fails to find the right DC (as opposed to `dal13`)

Need to lowercase the input so it matches what we get from the API.

```suggestion
            if datacenter and dc_name != datacenter.lower():
```

image

**Review by allmightyspiff — CHANGES_REQUESTED** [review]
Make sure to lowercase user input for datacenter here. Otherwise this looks great and ready to merge. Thanks.

**Review by allmightyspiff — APPROVED** [review]
Thanks, changes look good.

**Pull request merged**